libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/route/line/api/MapboxRouteLineView.kt

```diff
@@ -1106,24 +1110,7 @@ class MapboxRouteLineView(options: MapboxRouteLineOptions) {
     }
 
     private fun resetLayers(style: Style) {
-        sourceToFeatureMap.clear()
-        sourceToFeatureMap[MapboxRouteLineUtils.layerGroup1SourceKey] = RouteLineFeatureId(null)
-        sourceToFeatureMap[MapboxRouteLineUtils.layerGroup2SourceKey] = RouteLineFeatureId(null)
-        sourceToFeatureMap[MapboxRouteLineUtils.layerGroup3SourceKey] = RouteLineFeatureId(null)
-        primaryRouteLineLayerGroup = setOf()
-        listOf(
-            RouteLayerConstants.LAYER_GROUP_1_SOURCE_ID,
-            RouteLayerConstants.LAYER_GROUP_2_SOURCE_ID,
-            RouteLayerConstants.LAYER_GROUP_3_SOURCE_ID,
-            RouteLayerConstants.WAYPOINT_SOURCE_ID
-        ).forEach {
-            updateSource(
-                style,
-                it,
-                FeatureCollection.fromFeatures(listOf())
-            )
-        }
-        MapboxRouteLineUtils.removeLayersAndSources(style)
+        MapboxRouteLineUtils.removeLayers(style)
```

Comment:
Is my understanding correct that when a layer is readded, we'll have visible routes?
What about a case when we readd them and the routes are outdated? Is this possible?

Reply:
No, this operation is only done once for a fresh instance of the `MapboxRouteLineView`, to ensure that the applied `MapboxRouteLineOptions` are in sync with existing sources/layers. We don't run this logic for any other continuous updates.

Reply:
> to ensure that the applied MapboxRouteLineOptions are in sync with existing sources/layers

Do you mean the case when sources/layers with the same ids are created, but then we recreate `MapboxRouteLineView`? And here we want to make sure that all the sources/layers are "clean"?